pkgs/by-name/do/double-entry-generator/package.nix

```diff
@@ -0,0 +1,28 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+buildGoModule rec {
+  pname = "double-entry-generator";
+  version = "2.7.1";
+  src = fetchFromGitHub {
+    owner = "deb-sig";
+    repo = "double-entry-generator";
+    hash = "sha256-2Y8Spj1LAVZsUgChDYDCZ63pTH+nqs2ff9xcmC+gr0c=";
+    rev = "v${version}";
+  };
+
+  vendorHash = "sha256-Xedva9oGteOnv3rP4Wo3sOHIPyuy2TYwkZV2BAuxY4M=";
+
+  postInstall = ''
+    rm $out/bin/hack
+  '';
+
```

Comment:
Is it better to use `SHELL=${stdenv.shell}`?

Reply:
Other options are `SHELL=${pkgs.runtimeShell}`, `SHELL=bash` and `SHELL=sh`.  I am not sure which one is better though.

Reply:
`/bin/sh` just works, but the upstream uses `/bin/bash`. Is `/bin/sh` in Nix build sandbox a symlink to bash?

Reply:
At least on my environment (aarch64-darwin) it is bash.

```console
$ /bin/sh --version
GNU bash, version 3.2.57(1)-release (arm64-apple-darwin23)
Copyright (C) 2007 Free Software Foundation, Inc.
```

But I think it will be more consistent to use `bash`, in case the upstream can update the test scripts to reply on the new features of bash.

Reply:
> But I think it will be more consistent to use bash, in case the upstream can update the test scripts to reply on the new features of bash.

sh linked from bash is enough according to [this comment](https://github.com/deb-sig/double-entry-generator/blob/d28711bb65dd422c2155c03d9cbce4866ab74b88/Makefile#L21).  However, I agree that using bash is safer.

Reply:
After discussion with others, I find that `/bin/sh` should be avoided since it is a busybox sh on Linux and `pkgs.runtimeShell` is incorrect for cross compilation.  As a result, `SHELL=bash`, `SHELL=sh` and `SHELL={stdenv.shell}` are all correct.